service/src/amqp.ts

```diff
@@ -209,26 +209,27 @@ export class Amqp extends EventEmitter implements Client.Transport {
 
                 if (this._config.sharedAccessSignature) {
                   const audience = SharedAccessSignature.parse(this._config.sharedAccessSignature.toString(), ['sr', 'sig', 'se']).sr;
-                  const applicationSuppliedSas = typeof (this._config.sharedAccessSignature) === 'string';
+                  const isApplicationSuppliedSas = typeof (this._config.sharedAccessSignature) === 'string';
 
-                  const sasToken = applicationSuppliedSas ? this._config.sharedAccessSignature as string : (this._config.sharedAccessSignature as SharedAccessSignature).extend(anHourFromNow());
+                  const sasToken = isApplicationSuppliedSas ? this._config.sharedAccessSignature as string : (this._config.sharedAccessSignature as SharedAccessSignature).extend(anHourFromNow());
                   this._amqp.putToken(audience, sasToken, (err) => {
                     if (err) {
                       /*Codes_SRS_NODE_IOTHUB_SERVICE_AMQP_06_004: [** If `putToken` is not successful then the client will remain disconnected and the callback, if provided, will be invoked with an error object.]*/
                       this._fsm.transition('disconnecting', err, callback);
                     } else {
-                      this._fsm.transition('authenticated', applicationSuppliedSas, callback);
+                      this._fsm.transition('authenticated', isApplicationSuppliedSas, callback);
                     }
                   });
                 } else if (this._config.tokenCredential) {
-                  const audience = this._iotHubPublicScope;
                   this.getToken().then((accessToken) => {
-                    this._amqp.putToken(audience, accessToken, (err) => {
+                    const tokenValue = this._bearerTokenPrefix + accessToken.token;
+                    this._amqp.putToken(this._iotHubPublicScope, tokenValue, (err) => {
                       if (err) {
                         /*Codes_SRS_NODE_IOTHUB_SERVICE_AMQP_06_004: [** If `putToken` is not successful then the client will remain disconnected and the callback, if provided, will be invoked with an error object.]*/
                         this._fsm.transition('disconnecting', err, callback);
                       } else {
-                        this._fsm.transition('authenticated', accessToken, callback);
+                        this._renewalNumberOfMilliseconds = (2 / 3) * (accessToken.expiresOnTimestamp - Date.now());
```

Comment:
Do some serious thinking about this _renewalNumberOfMilliseconds.  Can you think of any reason for it to go negative OR zero?  I would hate to get into a tight renewal loop.

Reply:
Done